custom_components/adaptive_lighting/__init__.py

```diff
@@ -55,6 +57,13 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
     """Set up the component."""
     data = hass.data.setdefault(DOMAIN, {})
 
+    # This will reload any changes the user made to any YAML configurations.
+    # Called during 'quick reload' or hass.reload_config_entry
+    hass.bus.async_listen(
+        "hass.config.entry_updated",
+        lambda event: reload_configuration_yaml(event, hass),
```

Comment:
Are you sure this works? Doesn't `hass.bus.async_listen` expect an `async` function.

It might not, but it so, you can just define 

```python
    async def reload_configuration_yaml(event: dict):
        """Reload configuration.yaml."""
        await hass.services.async_call("homeassistant", "check_config", {})

    hass.bus.async_listen(
        "hass.config.entry_updated", reload_configuration_yaml,
    )
```

Reply:
it did work, sorry I was messing around with pre-commit to automatically type everything.

Reply:
Yeah as long as the listener is setup it'll work correctly. We should really move away from yaml configurations though, most are migrating to the new format. see https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision

Reply:
I really do not want to move away from YAML. I think UI makes sense for managing 1 or a couple of settings, but once there are many, YAML is so much nicer. Especially when managing multiple instances.